subprojects/architecture-test/src/test/java/org/gradle/architecture/test/ArchUnitFixture.java

```diff
@@ -69,18 +69,26 @@
 import static com.tngtech.archunit.core.domain.JavaModifier.PUBLIC;
 import static com.tngtech.archunit.core.domain.properties.HasModifiers.Predicates.modifier;
 import static com.tngtech.archunit.core.domain.properties.HasName.Functions.GET_NAME;
+import static com.tngtech.archunit.core.domain.properties.HasName.Predicates.nameMatching;
 import static com.tngtech.archunit.core.domain.properties.HasType.Functions.GET_RAW_TYPE;
 import static java.util.stream.Collectors.toSet;
 
 public interface ArchUnitFixture {
-    DescribedPredicate<JavaMember> public_api_methods = declaredIn(gradlePublicApi())
-        .and(modifier(PUBLIC))
-        .as("public API methods");
-
     DescribedPredicate<JavaMember> not_written_in_kotlin = declaredIn(
             resideOutsideOfPackages("org.gradle.configurationcache..", "org.gradle.kotlin.."))
             .as("API written in Java");
 
+    DescribedPredicate<JavaMember> kotlin_internal_methods = declaredIn(gradlePublicApi())
+        .and(not(not_written_in_kotlin))
+        .and(modifier(PUBLIC))
+        .and(nameMatching(".+\\$[a-z_]+"))
+        .as("Kotlin internal methods");
```

Comment:
🤔 I think this condition is hard to understand. Are these supposed to be the methods in the Kotlin packages which have a `$` in their method name?

Reply:
I added an explaining comment